joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -279,7 +284,7 @@ controller_interface::return_type JointTrajectoryController::update(
             tmp_command_[i] = (state_desired_.velocities[i] * ff_velocity_scale_[i]) +
                               pids_[i]->computeCommand(
                                 state_error_.positions[i], state_error_.velocities[i],
-                                (uint64_t)period.nanoseconds());
+                                (uint64_t)update_period_.nanoseconds());
```

Comment:
shouldn't this use the period itself?
I think the `period` is more correct than the `update_period_`

Reply:
It would usually not make that big of a difference, but isn't the goal here, as well, to generate a command to reach `state_desired` at the end of the current control cycle (which will be `update_period_` in the future).

Reply:
Actually, I agree with Sai here: This is not related to the trajectory sampling but only for calculating the feedback control law (in fact, only for integrating the error for the integral part of the command).